pkg/broker/filter/filter_handler_test.go

```diff
@@ -647,25 +645,33 @@ func makeTrigger(filter *eventingv1.TriggerFilter) *eventingv1.Trigger {
 			Name:      triggerName,
 			UID:       triggerUID,
 		},
-		Spec: eventingv1.TriggerSpec{
-			Filter: filter,
-		},
+		Spec: eventingv1.TriggerSpec{},
 		Status: eventingv1.TriggerStatus{
 			SubscriberURI: &apis.URL{Host: "toBeReplaced"},
 		},
 	}
+	for _, opt := range options {
+		opt(t)
+	}
+	return t
 }
 
-func makeTriggerWithoutFilter() *eventingv1.Trigger {
-	t := makeTrigger(makeTriggerFilterWithAttributes("", ""))
-	t.Spec.Filter = nil
-	return t
+func withUID(uid string) TriggerOption {
+	return func(t *eventingv1.Trigger) {
+		t.ObjectMeta.UID = types.UID(uid)
+	}
 }
 
-func makeTriggerWithoutSubscriberURI() *eventingv1.Trigger {
-	t := makeTrigger(makeTriggerFilterWithAttributes("", ""))
-	t.Status = eventingv1.TriggerStatus{}
-	return t
+func withAttributesFilter(filter *eventingv1.TriggerFilter) TriggerOption {
+	return func(t *eventingv1.Trigger) {
+		t.Spec.Filter = filter
+	}
+}
+
+func withoutSubscriberURI() TriggerOption {
+	return func(t *eventingv1.Trigger) {
+		t.Status = eventingv1.TriggerStatus{}
+	}
 }
```

Comment:
This is misleading and it might unexpectedly remove previously set conditions since it doesn't only set `SubscriberURI = nil` but it resets the entire status...

Reply:
ack 👍🏽 good catch. I was also surprised when I saw it then it got lost during the refactoring